pkg/clockcache/cache.go

```diff
@@ -95,11 +102,15 @@ func (self *Cache) insert(key interface{}, value interface{}) (canonicalKey inte
 		self.elementsLock.Lock()
 		defer self.elementsLock.Unlock()
 		delete(self.elements, insertLocation.key)
```

Comment:
We can't do that because the map key is changing

Reply:
Oh yes, I missed that. That makes the demand of a reduceable cache more. (Earlier we had issues in high churn when we wanted a reduce in cache size on my proposal, and now keys are never cleared, which makes the case more stronger). We can move forward for now but i think this may be important in future.

Reply:
Can you please add this in todo? We should really not miss this. @cevian